### PR TITLE
LPD-24353 LPD-24331 Bugs with a11y and facet terms on search-web templates

### PR DESCRIPTION
**Pull request by oliv-yu**
https://liferay.atlassian.net/browse/LPD-24353 - Accessibility error on Category Facet's Vocab Layout template
- Address a11y issue how user can tabulate between checkbox and description
- Add keypress function to toggle the tree item with keyboard

https://liferay.atlassian.net/browse/LPD-24331 - User facet template (compact, label) is not filtering properly due to incorrect data term id
- Not sure if anyone ever noticed, but the user facet templates for compact and label were filtering by the name but not the ID, so just a short update to change it to the filter value 😓

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web/src/main/resources/com/liferay/portal/search/web/internal/category/facet/portlet/display/template/dependencies/portlet_display_template_vocabulary.ftl [hunk 1]
```diff
@@ -186,4 +187,12 @@
 			}
 		}
 	}
+
+	function ${namespace}toggleTreeItemKeypress(event) {
+		event.preventDefault();
+
+		if (event.code === 'Enter') {
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/com/liferay/portal/search/web/internal/category/facet/portlet/display/template/dependencies/portlet_display_template_vocabulary.ftl:194]: `Space` should also toggle the tree item.

**Review by thektan — CHANGES_REQUESTED** [review]
Just 1 small addition and it can be forwarded afterwards!

**Comment by oliv-yu** [comment]
ci:forward

**Pull request closed without merging**